File: Leetcode/2125 Number of Laser Beams in a Bank.py
class Solution:
    def numberOfBeams(self, bank):
        last_non_zero_lasers = 0
        n_beams = 0
        for row in bank:
            lasers = sum(map(int, row))
            if lasers != 0:
                n_beams += lasers * last_non_zero_lasers
                last_non_zero_lasers = lasers
        return n_beams

File: Leetcode/2125 Number of Laser Beams in a Bank.ts
function numberOfBeams(bank: string[]): number {
    let lastNonZeroLasers = 0;
    let nBeams = 0;
    for (const row of bank) {
        let lasers = 0
        for (const c of row) {
            if (c === '1') {
                lasers++;
            }
        }
        if (lasers !== 0) {
            nBeams += lastNonZeroLasers * lasers;
            lastNonZeroLasers = lasers;
        }
    }
    return nBeams;
}
